clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -57,15 +58,22 @@
 
     public static final int AUTHORIZED_OPERATIONS_OMITTED = Integer.MIN_VALUE;
 
-    private MetadataResponseData data;
+    private final MetadataResponseData data;
```

Comment:
If we're going to go down the route of wrapping everything, then we should hide the `MetadataResponseData` field so that it can't be modified to be different than the wrapped fields.  So we shouldn't have `data` as a public method (it should be private, or absent)

Reply:
This is used in a couple of places in `KafkaAdminClient` currently.

Reply:
I replaced this with two methods that return an int so that callers cannot mutate `data`.